main.tf

```diff
@@ -200,86 +331,31 @@ resource "azurerm_private_dns_a_record" "arecord-1" {
   }
 }
 
-resource "azurerm_user_assigned_identity" "example" {
-  count               = var.enabled ? 1 : 0
-  resource_group_name = var.resource_group_name
-  location            = var.location
-  name                = format("midd-keyvault-%s", module.labels.id)
-  tags                = module.labels.tags
-}
-
-
-resource "azurerm_role_assignment" "aks_user_assigned" {
-  count                = var.enabled ? 1 : 0
-  principal_id         = join("", azurerm_user_assigned_identity.example.*.principal_id)
-  scope                = join("", azurerm_key_vault.key_vault.*.id)
-  role_definition_name = "Reader"
-}
-
-
-resource "azurerm_role_assignment" "rbac_user_assigned" {
-  count                = var.enabled && var.enable_rbac_authorization ? length(var.principal_id) : 0
-  principal_id         = element(var.principal_id, count.index)
-  scope                = join("", azurerm_key_vault.key_vault.*.id)
-  role_definition_name = element(var.role_definition_name, count.index)
-}
-
-resource "azurerm_key_vault_key" "example" {
-  depends_on   = [azurerm_key_vault.key_vault, ]
-  count        = var.enabled && var.key_enabled ? 1 : 0
-  name         = format("mid-keyvault-%s", module.labels.id)
-  key_vault_id = join("", azurerm_key_vault.key_vault.*.id)
-  key_type     = "RSA"
-  key_size     = 2048
-
-
-
-  key_opts = [
-    "decrypt",
-    "encrypt",
-    "sign",
-    "unwrapKey",
-    "verify",
-    "wrapKey",
-  ]
-  tags = module.labels.tags
-}
-
+##----------------------------------------------------------------------------- 
+## Below resources will create diagnostic setting for key vault and its components. 
+##-----------------------------------------------------------------------------
 resource "azurerm_monitor_diagnostic_setting" "example" {
```

Comment:
can we change use meaningful reference rather than using `example`

Reply:
we cannot provide a details about each resource. Like diagnosis setting is used for reason and other, if someone is using any resource he/she should has idea about diagnosis setting.